ironfish/src/memPool/memPool.ts

```diff
@@ -75,7 +94,9 @@ export class MemPool {
     let deletedTransactions = 0
 
     for (const transaction of block.transactions) {
-      this.transactions.delete(transaction.transactionHash())
+      const hash = transaction.transactionHash()
+      this.transactions.delete(hash)
+      this.queue.removeOne((t) => t.hash.equals(hash))
```

Comment:
this could be done faster with `queue.remove(value)`, there would be a bit more work to do that but we can talk through that.